### PR TITLE
Syscollector hotfixes missing

### LINKED ISSUE #9414: Some hotfixes not are displayed in kibana
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.0-RC11 | sysinfo | Agent | Packages/Sources | Windows |

## Description
During the validation tests of the performance improvement PR(#9394 ) for sysinfo in the hotfixes part, it was detected that there were missing elements to be displayed.

Steps to reproduce:
- Install Windows 10 21h1
- Run in powershell get-hotfix and keep hotfixes list
- Execute sysinfo_test_tool and compare the hotfix list with the previous steps.
image
image
image

Actual result: KB5003742 not appear in hotfix list.
Expected result: KB5003742 appear in hotfix list.

Possible RCA
The` rollupfix` field is ​​not being checked.
https://github.com/wazuh/wazuh/blob/0e1e79a070cb7775baa441d376a0bad3d65c14be/src/wazuh_modules/syscollector/syscollector_windows.c#L1000

## DoD
- [x] RCA and post it in this bug.
- [x] Implement fix.
- [x] Add unit test.
- [x] Modify architecture diagram
- [x] Run UMDH and analyze it to check leaks.
- [x] Run with APP Verifier to check memory issues.
- [x] E2E Functional test.

### PR DESCRIPTION
**Pull request by konangc87**
|Related issue|
|---|
| Closes #9414 |

## Description
This PR aims to fix the issue found during dev testing in issue #9386 related to some missing hotfixes in the items reported by sysinfo.

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [ ] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

**Review by tinachoB — APPROVED** [review]
LGTM!

File: src/data_provider/src/packages/packagesWindowsParserHelper.h [hunk 1]
```diff
@@ -43,28 +44,39 @@ namespace PackageWindowsHelper
...
-                if (Utils::startsWith(package, "Package_"))
+                [&key, &subKey, &hotfixes](const std::string& package)
                 {
-                    std::string value;
-                    Utils::Registry packageReg{key, subKey + "\\" + package, KEY_WOW64_64KEY | KEY_READ};
-                    if (packageReg.string("InstallLocation", value))
+                    if (Utils::startsWith(package, "Package_"))
                     {
-                        const auto hfValue { extractHFValue(value) };
+                        auto hfValue { extractHFValue(package) };
```
Comment (Dwordcito) [inline src/data_provider/src/packages/packagesWindowsParserHelper.h:53]: Based on https://github.com/wazuh/wazuh/blob/0e1e79a070cb7775baa441d376a0bad3d65c14be/src/wazuh_modules/syscollector/syscollector_windows.c#L1002 the block from line 53 to 57 should be in the else of the conditional on line 58 ?

**Review by Dwordcito — CHANGES_REQUESTED** [review]

File: src/data_provider/src/packages/packagesWindowsParserHelper.h (hunk 1 above)
Comment (konangc87) [reply src/data_provider/src/packages/packagesWindowsParserHelper.h:53]: flow is:
1- try to extract KB value from package name.
2- if 1 failed and the package name contains "RollupFix", try to extract the KB value from "InstallLocation" value.

**Comment by konangc87** [comment]
udmh logs:
[compare.txt](https://github.com/wazuh/wazuh/files/6908135/compare.txt)
[log2.txt](https://github.com/wazuh/wazuh/files/6908137/log2.txt)
[log1.txt](https://github.com/wazuh/wazuh/files/6908139/log1.txt)

**Comment by konangc87** [comment]
app verifier log:
[appverifier.txt](https://github.com/wazuh/wazuh/files/6908293/appverifier.txt)

**Review by Dwordcito — APPROVED** [review]

File: src/data_provider/src/packages/packagesWindowsParserHelper.h [hunk 2]
```diff
@@ -43,28 +44,39 @@ namespace PackageWindowsHelper
...
                         if (!hfValue.empty())
                         {
-                            hotfixes.insert(hfValue);
+                            hotfixes.insert(std::move(hfValue));
+                        }
+                        else if(package.find("RollupFix") != std::string::npos)
```
Comment (jnasselle) [inline src/data_provider/src/packages/packagesWindowsParserHelper.h:58]:
```suggestion
                        else if (package.find("RollupFix") != std::string::npos)
```

**Review by jnasselle — APPROVED** [review]

**Comment by jnasselle** [comment]
This PR was closed in favor of # 9414 due to the current repository privileges of the PR owner.

**Pull request closed without merging**